src/main/kotlin/net/atos/zac/app/informatieobjecten/InformationObjectRESTService.kt

```diff
@@ -0,0 +1,674 @@
+import net.atos.client.zgw.shared.util.URIUtil
+import net.atos.client.zgw.zrc.ZRCClientService
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.client.zgw.zrc.model.ZaakInformatieobject
+import net.atos.client.zgw.ztc.ZTCClientService
+import net.atos.client.zgw.ztc.util.InformatieObjectTypeUtil
+import net.atos.zac.app.audit.converter.RESTHistorieRegelConverter
+import net.atos.zac.app.audit.model.RESTHistorieRegel
+import net.atos.zac.app.informatieobjecten.converter.RESTInformatieobjectConverter
+import net.atos.zac.app.informatieobjecten.converter.RESTInformatieobjecttypeConverter
+import net.atos.zac.app.informatieobjecten.converter.RESTZaakInformatieobjectConverter
+import net.atos.zac.app.informatieobjecten.model.RESTDocumentCreatieGegevens
+import net.atos.zac.app.informatieobjecten.model.RESTDocumentCreatieResponse
+import net.atos.zac.app.informatieobjecten.model.RESTDocumentVerplaatsGegevens
+import net.atos.zac.app.informatieobjecten.model.RESTDocumentVerwijderenGegevens
+import net.atos.zac.app.informatieobjecten.model.RESTDocumentVerzendGegevens
+import net.atos.zac.app.informatieobjecten.model.RESTEnkelvoudigInformatieObjectVersieGegevens
+import net.atos.zac.app.informatieobjecten.model.RESTEnkelvoudigInformatieobject
+import net.atos.zac.app.informatieobjecten.model.RESTGekoppeldeZaakEnkelvoudigInformatieObject
+import net.atos.zac.app.informatieobjecten.model.RESTInformatieobjectZoekParameters
+import net.atos.zac.app.informatieobjecten.model.RESTInformatieobjecttype
+import net.atos.zac.app.informatieobjecten.model.RESTZaakInformatieobject
+import net.atos.zac.app.zaken.converter.RESTGerelateerdeZaakConverter
+import net.atos.zac.app.zaken.model.RelatieType
+import net.atos.zac.authentication.LoggedInUser
+import net.atos.zac.configuratie.ConfiguratieService
+import net.atos.zac.documentcreatie.DocumentCreatieService
+import net.atos.zac.documentcreatie.model.DocumentCreatieGegevens
+import net.atos.zac.documenten.InboxDocumentenService
+import net.atos.zac.documenten.OntkoppeldeDocumentenService
+import net.atos.zac.enkelvoudiginformatieobject.EnkelvoudigInformatieObjectLockService
+import net.atos.zac.event.EventingService
+import net.atos.zac.flowable.FlowableTaskService
+import net.atos.zac.flowable.TaakVariabelenService
+import net.atos.zac.policy.PolicyService
+import net.atos.zac.util.UriUtil
+import net.atos.zac.webdav.WebdavHelper
+import net.atos.zac.websocket.event.ScreenEventType
+import org.apache.commons.lang3.BooleanUtils
+import org.apache.commons.lang3.StringUtils
+import org.jboss.resteasy.annotations.providers.multipart.MultipartForm
+import java.io.IOException
+import java.net.URI
+import java.util.UUID
+
+@Singleton
+@Path("informatieobjecten")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@Suppress("TooManyFunctions", "LargeClass", "LongParameterList", "TooGenericExceptionThrown")
+class InformationObjectRESTService @Inject constructor(
```

Comment:
Do we really want to start translate ZGW terminology? So far we have translated internal ZAC stuff which is good and fine but this is a little different because an `enkelvoudiginformatieobject` is a ZGW term. If we go that way we will need to translate the entire ZGW Document API (https://vng-realisatie.github.io/gemma-zaken/standaard/documenten/) and I doubt we should do that..?

I would say: translate everything _except_ ZGW terminology. Also because it will make our code harder to read I'm afraid because the mapping to the ZGW APIs will then be less clear.

Reply:
Really, these ZGW terms do not have a place in our REST endpoints I would say. I would leave it for now because else we need to change everything but normally I would just call this thing our `DocumentRestService` or something? I mean the fact that the implementation uses the ZGW Documenten API under the hood should really be irrelevant for the API client (frontend in our case) I think. But oh well, one step at a time. :-)